demo-app/src/pages/api/synchronous-rest-query.ts

```diff
@@ -0,0 +1,27 @@
+import synchronousRestQuery from '@/lib/synchronous-rest-query'
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse,
+): Promise<void> {
+  if (request.method === 'POST') {
+    try {
+      const data =
+        request.headers['content-type'] === 'application/json'
+          ? request.body
+          : JSON.parse(request.body)
+      const result = await synchronousRestQuery(data)
+      // TODO: Should we proxy the status code instead of always returning 200?
```

Comment:
I think always returning 200 and including the actual status in the response body is best for now. This way `<OTRestDataProvider />` can differentiate between a failed call to `/api/synchronous-rest-query` and a failed call to the underlying REST source. And it can still choose how to use the actual status from the REST source.